daemon/job_actual_ip.go

```diff
@@ -0,0 +1,67 @@
+package daemon
+
+import (
+	"context"
+	"log"
+	"net/netip"
+	"time"
+
+	"github.com/NordSecurity/nordvpn-linux/core"
+	"github.com/NordSecurity/nordvpn-linux/internal"
+	"github.com/NordSecurity/nordvpn-linux/network"
+)
+
+func insightsUntilSuccess(ctx context.Context, api core.InsightsAPI) (core.Insights, error) {
+	for i := 0; ; i++ {
+		select {
+		case <-ctx.Done():
+			return core.Insights{}, ctx.Err()
+		default:
+			insights, err := api.Insights()
+			if err == nil && insights != nil {
+				return *insights, nil
```

Comment:
Is there a need to dereference here? It will copy whole insights struct. Maybe pointer will do? Or maybe we could just return the IP from here?